pygama/dsp/_processors/pulse_injector.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+from numba import guvectorize
+from math import log,exp
+
+@guvectorize(["void(float32[:], float32, float32, float32, float32, float32[:])",
+              "void(float64[:], float64, float64, float64, float64, float64[:])"],
+              "(n),(), (), (), ()->(n)", nopython=True, cache=True)
+def inject_sig_pulse(wf_in,t0,rise,amp,tau,wf_out):
+    """
+    Inject sigmoid pulse into existing waveform to simulate pileup
+    """    
+    wf_out[:] = wf_in[:]
+    for T in range(len(wf_out)):
+        wf_out[T] = wf_out[T]+ (amp/(1+exp((-1/rise)*(T-(t0+rise/2))))*exp(-(1/tau)*(T-t0)))

```

Comment:
@SeanSullivan1, I think line 12 is not needed. You can use directly wf_in at line 14. Same in the other function